### ISSUE TITLE
`ModelConfig.seed` and `accumulation_steps` are defined but silently ignored — reproducibility is impossible

### ISSUE THREAD

**Issue by bekucukakin** [dataxid/dataxid-python#33]
## Bug Description

`ModelConfig` has 16 public fields. Two of them — `seed` and `accumulation_steps` — are **defined in the dataclass but never read by any part of the SDK**. They are dead API surface. Setting `seed=42` has zero effect: every call produces different outputs, making reproducibility and deterministic testing impossible.

## Steps to Reproduce

```python
import dataxid
import pandas as pd

dataxid.api_key = "dx_test_..."

df = pd.DataFrame({"x": list(range(30)), "y": [i % 3 for i in range(30)]})
config = dataxid.ModelConfig(seed=42)

result1 = dataxid.synthesize(data=df, n_samples=10, config=config)
result2 = dataxid.synthesize(data=df, n_samples=10, config=config)

print(result1.equals(result2))  # Expected: True — Actual: False (always)
```

**Expected:** Same seed → identical synthetic output.

**Actual:** Results differ on every run. `seed` is completely ignored.

## Confirmed Test Run (dataxid 0.1.0, Python 3.12, Linux)

Verified via **source code inspection** (no API call required):

```
✗ FAIL  'seed' is defined in ModelConfig but never read anywhere!
        Ignored fields: ['accumulation_steps', 'seed']
```

The inspection checked every `config["..."]` read in `Model.create()` and `train_frozen()` against all 16 `ModelConfig` dataclass fields. Only 14 fields are actually used.

### `ModelConfig` definition (`training/_config.py`)

All 16 fields are declared:

```python
@dataclass
class ModelConfig:
    embedding_dim: int = 64
    model_size: Literal["small", "medium", "large"] = "medium"
    batch_size: int = 256
    max_epochs: int = 100
    early_stop_patience: int = 4
    val_split: float = 0.1
    privacy_enabled: bool = False
    privacy_noise: float = 0.1
    encoding_types: dict[str, str] | None = None
    accumulation_steps: int = 1        # ← NEVER READ
    learning_rate: float | None = None
    label_smoothing: float = 0.0
    embedding_dropout: float = 0.5
    time_limit_seconds: float = 0.0
    seed: int | None = None            # ← NEVER READ
    timeout: float = 14400.0
```

### `Model.create()` reads only 9 fields (line 26–34):

```python
embedding_dim         = config["embedding_dim"]
model_size            = config["model_size"]
batch_size            = config["batch_size"]
max_epochs            = config["max_epochs"]
early_stop_patience   = config["early_stop_patience"]
privacy_enabled       = config["privacy_enabled"]
privacy_noise         = config["privacy_noise"]
encoding_types        = config["encoding_types"]
val_split             = config["val_split"]
# seed and accumulation_steps are NEVER READ HERE
```

### `train_frozen()` sends this payload to `/v1/models/{id}/init-training`:

```python
model._client.post(f"/v1/models/{model.id}/init-training", json={
    "max_epochs":         max_epochs,
    "patience":           early_stop_patience,
    "learning_rate":      initial_lr,
    "label_smoothing":    model._config["label_smoothing"],
    "embedding_dropout":  model._config["embedding_dropout"],
    "time_limit_seconds": model._config["time_limit_seconds"],
    "batches":            batches,
    # "seed" is NOT included here
    # "accumulation_steps" is NOT included here
})
```

`seed` is read nowhere — not in `Model.create()`, not in `train_frozen()`, not in `Encoder.__init__()`. `accumulation_steps` is similarly defined but never consumed.

## Impact

| Field | Defined | Actually Used | Effect |
|-------|---------|---------------|--------|
| `seed` | ✅ | ❌ | Setting it does nothing; runs are always non-deterministic |
| `accumulation_steps` | ✅ | ❌ | Gradient accumulation never applied; ignored silently |

Reproducibility is a core requirement for:
- Regression testing of synthetic data pipelines
- Auditing and compliance workflows
- Debugging model behavior

## Fix Suggestion

**Server-side:** Pass `seed` in the `/init-training` payload:

```python
model._client.post(f"/v1/models/{model.id}/init-training", json={
    ...
    "seed": model._config.get("seed"),
    "accumulation_steps": model._config.get("accumulation_steps", 1),
})
```

**Client-side:** Apply seed to local PyTorch/NumPy operations before encoding:

```python
# In train_frozen(), before encoder.freeze():
seed = model._config.get("seed")
if seed is not None:
    import torch, numpy as np, random
    torch.manual_seed(seed)
    np.random.seed(seed)
    random.seed(seed)
```

**At minimum:** Raise a `UserWarning` if `seed` is set, so the user knows it has no effect:

```python
if config.get("seed") is not None:
    import warnings
    warnings.warn(
        "ModelConfig.seed is not yet implemented and will be ignored.",
        UserWarning,
        stacklevel=3,
    )
```

## Environment

- **OS:** Linux (Ubuntu 22.04)
- **Python:** 3.12
- **dataxid:** 0.1.0
